### PR TITLE
Fixes NullReferenceException in SimpleLayout PreCalculate after TryGetRawValue

### PR DESCRIPTION
**Pull request by snakefoot**
Fixed bug introduced with #3144

Note merge to master

### CODE REVIEW

File: src/NLog/Layouts/SimpleLayout.cs [hunk 1]
```diff
@@ -306,13 +306,12 @@ public override void Precalculate(LogEventInfo logEvent)
                 {
                     // If raw value doesn't have the ability to mutate, then we can skip precalculate
                     var success = _rawValueRenderer.TryGetRawValue(logEvent, out var value);
-                    if (success && Convert.GetTypeCode(value) != TypeCode.Object || value.GetType().IsValueType())
+                    if (success && (Convert.GetTypeCode(value) != TypeCode.Object || value.GetType().IsValueType()))
```
Comment (304NotModified) [inline src/NLog/Layouts/SimpleLayout.cs]: proposal, check on `null` explicitly:

```
  if (success && value != null && (Convert.GetTypeCode(value) != TypeCode.Object || value.GetType().IsValueType()))
```

File: src/NLog/Layouts/SimpleLayout.cs [hunk 2]
```diff
@@ -323,15 +322,15 @@ internal override void PrecalculateBuilder(LogEventInfo logEvent, StringBuilder
...
-
-            try
+            if (_rawValueRenderer != null)
             {
-                if (_rawValueRenderer != null)
+                try
                 {
+                    if (!IsInitialized)
+                    {
+                        Initialize(LoggingConfiguration);
```
Comment (304NotModified) [inline src/NLog/Layouts/SimpleLayout.cs]: Initialize on Precalculate hasn't a try-catch either?

File: src/NLog/Layouts/SimpleLayout.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog/Layouts/SimpleLayout.cs]: `Convert.GetTypeCode(null)` returns [TypeCode.Empty](https://docs.microsoft.com/en-us/dotnet/api/system.typecode), so extra check not needed.

File: src/NLog/Layouts/SimpleLayout.cs (hunk 2 above)
Comment (snakefoot) [reply src/NLog/Layouts/SimpleLayout.cs]: Fixed

File: src/NLog/Layouts/SimpleLayout.cs (hunk 1 above)
Comment (304NotModified) [reply src/NLog/Layouts/SimpleLayout.cs]: it's needed for the readability ;)

It's a bit hidden that there isn't a nullref exception with `value.GetType()` in this case. Also the overhead of the method call is a bit of waste

Comment (304NotModified) [reply src/NLog/Layouts/SimpleLayout.cs]: added it :angel:

**Pull request merged**